// karma/lib/reporters/Brackets.js
'use strict';

var io = require('socket.io-client');



var BracketsReporter = function (config) {

    // console.log('helper:', config);
    //

    // var optionsFor = function(type, browser) {
    //   return helper.merge(OPTIONS[type], {title: util.format(OPTIONS[type].title, browser)});
    // };

    // growly.register('Karma', '', [], function(error) {
    //   var warning = 'No running verion of GNTP found.\n' +
    //                 'For more information see https://github.com/theabraham/growly.';
    //   if (error) {
    //     log.warn(warning);
    //   }
    // });

    var sock = io.connect('http://localhost:5000');

    sock.on('error', function () {
        console.warn('Unable to connect to http://localhost:5000');
    });

    var socket = {
        emit: function (msg, data) {
            // console.warn('no server found, skipping');
            this.queue.push({
                msg: msg,
                data: data
            });
        },
        queue: []
    };
    sock.on('connect', function () {
        console.log('Connesso!!');
        if (socket.queue.length > 0) {
            socket.queue.forEach(function (item) {
                sock.emit(item.msg, item.data);
            });
            socket.queue = [];
        }
        socket = sock;

    });
    this.renderBrowser = function (browser) {
        console.log('renderBrowser');
        socket.emit('renderBrowser', browser);
        // console.log(browser);
    };

    this.onBrowserError = function (browser, error) {
        console.log('onBrowserError');
        socket.emit('browserError', {
            browser: browser,
            error: error
        });
        // console.log(browser);
    };

    this.onSpecComplete = function (browser, result) {
        console.log('onSpecComplete');
        socket.emit('specComplete', {
            browser: browser,
            result: result
        });
        // console.log(result);
        // console.log(browser);
    };

    this.onBrowserComplete = function (browser) {

        console.log('onBrowserComplete');
        socket.emit('browserComplete', {
            browser: browser
        });
        // console.log(browser);
        // var results = browser.lastResult;
        // // var time = helper.formatTimeInterval(results.totalTime);

        // if (results.disconnected || results.error) {
        //     return console.log('Ueeeeee');
        // };

        // if (results.failed) {
        //     console.log(':(');
        // };

        // console.log(':)');
    };

    this.onRunComplete = function (browsers, results) {
        // console.log(results);
        console.log('onRunComplete');
        socket.emit('runComplete', {
            browsers: browsers,
            results: results
        });

    };
    this.adapters = [];


};

// PUBLISH DI MODULE
module.exports = BracketsReporter;
